### PR TITLE
dev: update return type for get_logs method

### PR DESCRIPTION
**Pull request by aniketpr01**
Time spent on this PR: 3

Resolves: #440 

# Pull Request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Added changes to [`CHANGELOG.md`](../CHANGELOG.md)

- [x] Yes
- [ ] No

### CODE REVIEW

File: crates/core/src/client/tests/mod.rs [hunk 1]
```diff
@@ -248,41 +249,45 @@ async fn test_get_logs() {
...
+                ),
+                transaction_index: None,
+                log_index: None,
+                removed: false
+            },
+            logs[1]
+        )
+    } else {
+        panic!("Expected FilterChanges::Logs variant, got {:?}", logs);
+    }
```
Comment (greged93) [inline crates/core/src/client/tests/mod.rs:292]:
```suggestion
    match logs {
        FilterChanges::Logs(logs) => {
            assert_eq!(2, logs.len());
            assert_eq!(
                Log {
                    address: *ABDEL_ETHEREUM_ADDRESS,
                    topics: vec![],
                    data: Bytes::from_str("0xdead").unwrap(),
                    block_hash: Some(
                        H256::from_str("0x0197be2810df6b5eedd5d9e468b200d0b845b642b81a44755e19047f08cc8c6e").unwrap()
                    ),
                    block_number: Some(U256::from(5u8)),
                    transaction_hash: Some(
                        H256::from_str("0x032e08cabc0f34678351953576e64f300add9034945c4bffd355de094fd97258").unwrap()
                    ),
                    transaction_index: None,
                    log_index: None,
                    removed: false
                },
                logs[0]
            );
            assert_eq!(
                Log {
                    address: *ABDEL_ETHEREUM_ADDRESS,
                    topics: vec![],
                    data: Bytes::from_str("0xbeef").unwrap(),
                    block_hash: Some(
                        H256::from_str("0x0197be2810df6b5eedd5d9e468b200d0b845b642b81a44755e19047f08cc8c6e").unwrap()
                    ),
                    block_number: Some(U256::from(5u8)),
                    transaction_hash: Some(
                        H256::from_str("0x01b7ec62724de1faba75fdc75cf11c1f855af33e4fe5f36d8a201237f3c9f257").unwrap()
                    ),
                    transaction_index: None,
                    log_index: None,
                    removed: false
                },
                logs[1]
            )
        }
        _ => panic!("Expected FilterChanges::Logs variant, got {:?}", logs),
    }
```

File: crates/core/tests/client.rs [hunk 2]
```diff
@@ -146,46 +146,52 @@ mod tests {
         let events = client.get_logs(filter).await.unwrap();
```
Comment (greged93) [inline crates/core/tests/client.rs:146]:
```suggestion
        let logs = client.get_logs(filter).await.unwrap();
```

File: crates/core/tests/client.rs [hunk 3]
```diff
@@ -146,46 +146,52 @@ mod tests {
...
+                    ),
+                    transaction_index: None,
+                    log_index: None,
+                    removed: false
+                },
+                events[1]
+            );
+        } else {
+            panic!("Expected FilterChanges::Logs variant, got {:?}", events);
+        }
```
Comment (greged93) [inline crates/core/tests/client.rs:195]:
```suggestion
        match logs {
            FilterChanges::Logs(logs) => {
                assert_eq!(2, logs.len());
                assert_eq!(
                    Log {
                        address: erc20_eth_address,
                        topics: vec![
                            H256::from_str("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef")
                                .unwrap(), // keccak256(“Transfer(address,address,uint256)”)
                            H256::from_low_u64_be(0u64),                   // from
                            H256::from(kakarot.eoa_addresses.eth_address)  // to
                        ],
                        data: Bytes::from_str("0x0000000000000000000000000000000000000000000000000000000000002710")
                            .unwrap(), // amount
                        block_hash: logs[0].block_hash, // block hash changes so just set to event value
                        block_number: logs[0].block_number, // block number changes so just set to event value
                        transaction_hash: Some(
                            H256::from_str("0x0124c05cceb7e556f354f580a362845c746a2616d682ce3235c67e7b42a0fdd8")
                                .unwrap()
                        ),
                        transaction_index: None,
                        log_index: None,
                        removed: false
                    },
                    logs[0]
                );
                assert_eq!(
                    Log {
                        address: erc20_eth_address,
                        topics: vec![
                            H256::from_str("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef")
                                .unwrap(), // keccak256("Transfer(address,address,uint256)")
                            H256::from(kakarot.eoa_addresses.eth_address), // from
                            H256::from(*ACCOUNT_ADDRESS_EVM)               // to
                        ],
                        data: Bytes::from_str("0x0000000000000000000000000000000000000000000000000000000000002710")
                            .unwrap(), // amount
                        block_hash: logs[1].block_hash, // block hash changes so just set to event value
                        block_number: logs[1].block_number, // block number changes so just set to event value
                        transaction_hash: Some(
                            H256::from_str("0x00c2f52f03d1f8bc3995c533983364b077040093207c03393b4fd6b99e4af3ab")
                                .unwrap()
                        ),
                        transaction_index: None,
                        log_index: None,
                        removed: false
                    },
                    logs[1]
                );
            }
            _ => panic!("Expected FilterChanges::Logs variant, got {:?}", logs),
        }
```

**Review by greged93 — COMMENTED** [review]
Some small changes, also I prefer the `match` over the `if let` approach, might be personal but looks a little cleaner imo

File: crates/core/src/client/tests/mod.rs (hunk 1 above)
Comment (aniketpr01) [reply crates/core/src/client/tests/mod.rs]: Done!

File: crates/core/tests/client.rs (hunk 2 above)
Comment (aniketpr01) [reply crates/core/tests/client.rs]: Done!

File: crates/core/tests/client.rs (hunk 3 above)
Comment (aniketpr01) [reply crates/core/tests/client.rs]: Done!

File: crates/core/src/client/tests/mod.rs [hunk 4]
```diff
@@ -207,13 +207,14 @@ async fn test_get_logs_from_bigger_than_current() {
     let logs = client.get_logs(filter).await.unwrap();
 
     // Then
-    assert!(logs.is_empty());
+    assert!(matches!(logs, FilterChanges::Empty));
```
[the remark this answers is not in the document]
Comment (aniketpr01) [reply crates/core/src/client/tests/mod.rs]: Done!

File: crates/core/src/client/tests/mod.rs [hunk 5]
```diff
@@ -207,13 +207,14 @@ async fn test_get_logs_from_bigger_than_current() {
...
 }
 
 #[tokio::test]
 async fn test_get_logs_to_less_than_from() {
     // Given
-    let fixtures = fixtures(vec![wrap_kakarot!(JsonRpcMethod::BlockNumber)]);
+    // let fixtures = fixtures(vec![wrap_kakarot!(JsonRpcMethod::BlockNumber)]);
+    let fixtures = fixtures(vec![wrap_kakarot!(JsonRpcMethod::BlockNumber), wrap_kakarot!(JsonRpcMethod::GetEvents)]);
```
[the remark this answers is not in the document]
Comment (aniketpr01) [reply crates/core/src/client/tests/mod.rs:217]: Done!

[the remark this answers is not in the document]
Comment (greged93) [reply crates/core/src/client/tests/mod.rs:217]: I tried locally and this was fine, you shouldn't need the GetEvents fixture as it should stop on the first match of the `get_logs` function

[the remark this answers is not in the document]
Comment (aniketpr01) [reply crates/core/src/client/tests/mod.rs]: yes, retested and working fine, reverting!

**Comment by aniketpr01** [comment]
> Please just revert changes on the `test_get_logs_to_less_than_from` and update the change log.

Done!

File: crates/core/src/client/tests/mod.rs [hunk 6]
```diff
@@ -207,12 +207,13 @@ async fn test_get_logs_from_bigger_than_current() {
...
 
     // Then
-    assert!(logs.is_empty());
+    assert_eq!(logs, FilterChanges::Empty);
 }
 
 #[tokio::test]
 async fn test_get_logs_to_less_than_from() {
     // Given
+    // let fixtures = fixtures(vec![wrap_kakarot!(JsonRpcMethod::BlockNumber)]);
```
Comment (greged93) [inline crates/core/src/client/tests/mod.rs:216]: can you remove this comment please?

Comment (aniketpr01) [reply crates/core/src/client/tests/mod.rs]: yes ser! my apologies for ignoring this!

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**